### PR TITLE
Apply new design for the core profiler Jetpack connect page

### PR DESCRIPTION
**Pull request by moon0326**
Related to #92103 

This PR applies the new design for the "One last step!" page for both singed in & out.

This PR is going to serve as a base PR for the upcoming PRs for https://github.com/Automattic/wp-calypso/issues/92105

## Proposed Changes

Signed-in:

Screenshot 2024-07-31 at 2 02 12 PM

Signed-out:
Screenshot 2024-07-31 at 2 02 03 PM

## Why are these changes being made?

* Apply new design

## Testing Instructions

* Create a new JN site with the latest trunk
* Start the core profiler and choose 
* Make sure to check Jetpack on the extensions page and click on `Continue`. You'll be redirected to the Jetpack connect page.
* Copy the URL on the Jetpack connect page.
* Replace https://wordpress.com with your local Calypso url (http://calypso.localhost:3000/)
* Paste the URL.
* You should see the updated design.
* Open an incognito window and past the URL.
* You should see the updated design for signed-out users.

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

### CODE REVIEW

File: client/jetpack-connect/auth-form-header.jsx [hunk 1]
```diff
@@ -166,7 +167,7 @@ export class AuthFormHeader extends Component {
 			const pluginNames = {
 				'jetpack-ai': 'Jetpack AI',
 				'jetpack-boost': 'Jetpack Boost',
-				default: 'Jetpack',
+				default: 'Jetpack and WooPayments',
```
Comment (moon0326) [inline client/jetpack-connect/auth-form-header.jsx]: This should be dynamic depending on the chosen extensions. We should also make it customizable for other extensions. For now, let's go with hard-coded text, as we're not merging this PR until other changes are complete.

File: client/jetpack-connect/authorize.js [hunk 2]
```diff
@@ -75,7 +75,6 @@ import {
 	REMOTE_PATH_AUTH,
 } from './constants';
 import Disclaimer from './disclaimer';
-import { JetpackFeatures } from './features';
```
Comment (rjchow) [inline client/jetpack-connect/authorize.js]: I think the `./features` file can be deleted, there are no references to it

**Review by rjchow — COMMENTED** [review]
The TOS text on the signed-out screen seems different from the Figma design, should we clarify about that?

image

Jetpack plugin seems to trigger the Jetpack AI flow due to this: https://github.com/woocommerce/woocommerce/blob/a3f4d722b99a0384af4fa25abf6d607be270a5d9/plugins/woocommerce-admin/client/core-profiler/index.tsx#L316

It looks like a previously implemented design with just a difference in copy - just checking to see if that's intentional?

File: client/jetpack-connect/authorize.js (hunk 2 above)
Comment (moon0326) [reply client/jetpack-connect/authorize.js]: Good catch! Deleted it in [Remove unused features file](https://github.com/Automattic/wp-calypso/pull/93163/commits/0df4172b236b2072bfadcd3e584749d51ef73301)

Comment (moon0326) [reply client/jetpack-connect/authorize.js]: > Jetpack plugin seems to trigger the Jetpack AI flow due to this: https://github.com/woocommerce/woocommerce/blob/a3f4d722b99a0384af4fa25abf6d607be270a5d9/plugins/woocommerce-admin/client/core-profiler/index.tsx#L316

Good catch! I'll see if we're still running that test 🤔 I think i've removed it from other places.

Comment (moon0326) [reply client/jetpack-connect/authorize.js]: @rjchow

Update: We've removed the experiment in https://github.com/woocommerce/woocommerce/pull/49452 and I've removed `jetpack-boost` in https://github.com/Automattic/wp-calypso/pull/93163/commits/c398ba75b686b198a3d0f3929f9a578fda537249. This commit also sets `Jetpack and WooPayments` for `jetpack-ai`. I need to revisit how this plugin name replacement works later.

Comment (moon0326) [reply client/jetpack-connect/authorize.js:78]: I've also opened https://github.com/woocommerce/woocommerce/pull/50378

**Review by rjchow — APPROVED** [review]
Thanks for addressing the feedback!

File: client/jetpack-connect/auth-form-header.jsx (hunk 1 above)
Comment (psealock) [reply client/jetpack-connect/auth-form-header.jsx]: should it be translated?

**Review by psealock — APPROVED** [review]
Code looks good and this is testing well. Just one minor comment

**Comment by moon0326** [comment]
@psealock 

> should it be translated?

Good catch! Fixed it in [Translate plugin names](https://github.com/Automattic/wp-calypso/pull/93163/commits/46893a6b61d76d09de176209ac4f44350fead83a)

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/16713081

Some locales (Hebrew, Japanese) have been temporarily machine-translated due to translator availability. All other translations are usually ready within a few days. Untranslated and machine-translated strings will be sent for translation next Monday and are expected to be completed by the following Friday.

Thank you @moon0326 for including a screenshot in the description! This is really helpful for our translators.

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/16713081).

**Pull request merged**